pkg/apis/authorization/types.go

```diff
@@ -144,3 +144,53 @@ type SubjectAccessReviewStatus struct {
 	// For instance, RBAC can be missing a role, but enough roles are still present and bound to reason about the request.
 	EvaluationError string
 }
+
+// +genclient=true
+// +noMethods=true
+
+// SelfSubjectRulesReview enumerates the set of actions the current user can perform within a namespace.
+// The returned list of actions may be incomplete depending on the server's authorization mode,
+// and any errors experienced during the evaluation.
+type SelfSubjectRulesReview struct {
```

Comment:
A little confused by the SubjectAccessReview comparison. Can't users of SARs already control the user.Extra fields?

Wouldn't this be solved by a SubjectRulesReview (no Self) where the user can directly query any user? If extras are significant, it seems like I should have to go through some authorization kind of check (e.g. impersonation) before I can get information about that field.